eclair-node-gui/src/main/scala/fr/acinq/eclair/gui/utils/CoinFormat.scala

```diff
@@ -0,0 +1,28 @@
+package fr.acinq.eclair.gui.utils
+
+import java.text.{DecimalFormat, NumberFormat}
+import java.util.Locale
+
+object CoinFormat {
+  /**
+    * Always 3 decimals, maximum of 8
+    */
+  val BTC_PATTERN = "###,##0.000#####"
+
+  /**
+    * Always 5 decimals
+    */
+  val MILLI_BTC_PATTERN = "###,##0.00000"
+
+  /**
+    * Localized formatter for BTC amounts. Uses `BTC_PATTERN`.
+    */
+  val BTC_FORMAT: DecimalFormat = NumberFormat.getNumberInstance(Locale.getDefault).asInstanceOf[DecimalFormat]
```

Comment:
How can you be sure the returned `NumberFormat` is always a `DecimalFormat`?

Reply:
How about:

```scala
  val BTC_FORMAT: NumberFormat = NumberFormat.getNumberInstance(Locale.getDefault) match {
    case df: DecimalFormat => // should always be the case (see NumberFormatProviderImpl.java)
      df.applyPattern(BTC_PATTERN)
      df
    case other => other
  }
```

Reply:
Casting `NumberFormat` to `DecimalFormat` works for the vast majority of locale. I'll add a try catch for unusual ones.